backend/codingchallenge/api/views/applications.py

```diff
@@ -223,30 +235,114 @@ def delete(self, request, *args, **kwargs):
         application.delete()
         return Response(jsonMessages.successJsonResponse(), status=status.HTTP_200_OK)
 
+
+class AdminResultApplicationView(APIView):
+
+    permission_classes = [IsAuthenticated]
+
+    gApi = GithubApi()
+
+    # 8. Get Result
+    # https://github.com/ampcc/coding-challenge/wiki/API-Documentation-for-admin-functions#8-get-result
+    # /api/admin/applications/results/{applicationId}
+    # Todo: Test Cases for this method
+    def get(self, request, *args, **kwargs):
+        """
+        get Linter Results with
+            query:
+                applicationId
+        """
+        try:
+            application = Application.objects.get(applicationId=self.kwargs["applicationId"])
+
+            if not application:
+                raise TypeError
+
+        except(KeyError, TypeError):
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        if application.githubRepo:
+            repoName = application.githubRepo
+        else:
+            return Response(
+                jsonMessages.errorJsonResponse("Can not find repo"),
+                status=status.HTTP_400_BAD_REQUEST)
+
+
+        return Response({'content': self.gApi.getLinterResult(repoName)}, status=status.HTTP_200_OK)
+
+class UploadApplicationView(APIView):
+    permission_classes = [IsAuthenticated]
+    parser_classes = [FileUploadParser]
+
+    gApi = GithubApi()
+
+    # 18. Upload Challenge
+    # https://github.com/ampcc/coding-challenge/wiki/API-Documentation-for-applicant-functions#18-upload-challenge
+    # /api/application/uploadChallenge
+    # Todo: Test Cases for this method
+    def post(self, request, *args, **kwargs):
+        """
+        post Challenge with
+            required arguments:
+                dataZip
+        """
+        user = User.objects.get(username=request.user.username)
+
+        if user.application.status < Application.Status.IN_REVIEW:
+
+            repoName = f'{user.application.applicationId}_{user.application.challengeId}'
+
+            self.gApi.createRepo(repoName, 'to be defined')
+
+            raw_file = request.data['file']
+            file_obj = ZipFile(raw_file)
+
+            for path in file_obj.namelist():
+                if not path.endswith('/'):
+                    self.gApi.pushFile(repoName, path, file_obj.read(path))
+
+            self.gApi.addLinter(repoName)
+
+            user.application.submission = time.time()
+            user.application.status = Application.Status.IN_REVIEW
+            user.application.githubRepo = repoName
+            user.application.save()
+
+            return Response({"success": "true"}, status=status.HTTP_200_OK)
```

Comment:
use jsonMessages. ... instead of new success object

Reply:
done